### PR TITLE
refactor: Replace react-bootstrap Tabs with Antd Tabs in DashboardBuilder

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR replaces usages of `react-bootstrap`'s Tabs component with tabs from `antd` in DashboardBuilder. As you can see on the gifs, besides the new look I also removed the popover with delete button. Now it should be more intuitive.
On the downside, drag'n'drop functionality is a bit less handy - you need to drag a tab to the edge of another tab's title rather than the edge of tab itself. It worked the same in the old version, but due to different UI it was more noticeable. Changing this behaviour would require a major refactor - I think we should do it, but as a part of a different task/PR.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
before

After:
after

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
@rusackas Can you please take a look?

**Comment by mistercrunch** [comment]
Should we use the other style of antd tabs instead of these?

**Comment by kgabryje** [comment]
@mistercrunch I think it's a great idea to ask for a designer's input, but let's keep in mind that in the case of editable tabs (with add and remove options) we have no choice and we need to use the "card" style. However in the case of non-editable tabs we can use either "line" or "card" style.

**Comment by mistercrunch** [comment]
Depending on designers input, we may be able to switch the tab style in edit/view mode if necessary too. @kgabryje is it because the "non-card" tabs don't allow for arbitrary React node? They force you to pass a string?

**Comment by kgabryje** [comment]
Contrary to react-bootstrap tabs, here we use Antd's native functionality of adding and removing tabs, and it's required to pass `type=editable-card` for it to work. Of course we can switch styles between "line" and "card" depending on edit mode, but here I opted for keeping card style for consistency.

**Comment by rusackas** [comment]
@kgabryje I posed this question to designers (admittedly, internal ones... the broader community is welcome to engage here!), and got some feedback:

The technical constraints, requiring the use of the card-style tabs here are acknowledged. The design perspective, however, is that we should limit the number of _visible_ tab styles in our design system. The preferred approach would be to use these card-based tabs, but restyle them so they have a more consistent _appearance_ with the underlined tabs we use elsewhere.

I don't believe it should block this PR, as I think this PR is a big step in the right direction. A PR in which we re-style the various AntD tab flavors to be more consistent seems like a worthy follow-up task.

Curious how others feel about that proposal.

**Comment by ktmud** [comment]
Not a designer, but I also feel the original design is a cleaner and better user experience. It'd be a little weird to have different visual styles (line vs card) for tabs in View and in Edit modes.

In general, we should probably limit style changes when replacing Bootstrap components. For this case, it'd be easier and less risky if we had just kept the original editing functionalities and only replaced the `BootstrapTabs` component itself.

File: superset-frontend/stylesheets/superset.less
Comment (rusackas) [inline superset-frontend/stylesheets/superset.less:288]: Would much rather see this moved into Emotion, in the `Tabs` component, and using `gridUnit * 3` instead of `12`

File: superset-frontend/spec/javascripts/dashboard/components/gridComponents/Tabs_spec.jsx
Comment (rusackas) [inline superset-frontend/spec/javascripts/dashboard/components/gridComponents/Tabs_spec.jsx:111]: might be worth considering a data-test attribute here, in case our class names change for library/styling purposes.

File: superset-frontend/spec/javascripts/dashboard/components/gridComponents/Tabs_spec.jsx
Comment (rusackas) [inline superset-frontend/spec/javascripts/dashboard/components/gridComponents/Tabs_spec.jsx:66]: If you'd rather not import the `supersetTheme` and `ThemeProvider` to use here, you can instead `import { styledMount as mount } from 'spec/helpers/theming'` and then just do 
```
<Provider store={mockStoreWithTabs}>
  <WithDragDropContext>
    <Tabs {...props} {...overrideProps} />
  </WithDragDropContext>
</Provider>
```

**Comment by rusackas** [comment]
In the last conversation I had with designers about this, the conclusion was:
• We should use the same style of tabs in presentation and edit mode
• Therefore, if we need to use the card-based tabs in edit mode, then we should use that style of tabs in both modes.
• It would be _great_ if we can then _style_ those tabs to make them look like the underlined AntD tabs we're starting to add to other views. 
• If that styling change needs to happen in another PR, so we can get this one across the finish line, so be it.

Any objections to the above?

**Comment by kgabryje** [comment]
> In the last conversation I had with designers about this, the conclusion was:
> • We should use the same style of tabs in presentation and edit mode
> • Therefore, if we need to use the card-based tabs in edit mode, then we should use that style of tabs in both modes.
> • It would be _great_ if we can then _style_ those tabs to make them look like the underlined AntD tabs we're starting to add to other views.
> • If that styling change needs to happen in another PR, so we can get this one across the finish line, so be it.
> 
> Any objections to the above?

Thank you, I'll restyle the tabs to make them look more like "line" style tabs.

**Comment by kgabryje** [comment]
New look of the tabs in DashboardBuilder:
image

@rusackas @ktmud Thoughts?

**Comment by kgabryje** [comment]
I did a rebase to resolve conflicts. Also, I moved styled from `tabs.less` to Emotion and uppercased editable title input for consistency with other tabs modes.

**Review by suddjian — APPROVED** [review]
looks good to me, happy to merge if others' concerns are addressed.

**Review by rusackas — APPROVED** [review]
LGTM

**Pull request merged**
